File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/consensus/ProposeValue.java
package com.github.dedis.popstellar.model.network.method.message.data.consensus;

import com.github.dedis.popstellar.model.Immutable;
import com.google.gson.annotations.SerializedName;

import java.util.Objects;

@Immutable
public final class ProposeValue {

  @SerializedName("proposed_try")
  private final int proposedTry;

  @SerializedName("proposed_value")
  private final boolean proposedValue;

  public ProposeValue(int proposedTry, boolean proposedValue) {
    this.proposedTry = proposedTry;
    this.proposedValue = proposedValue;
  }

  public int getProposedTry() {
    return proposedTry;
  }

  public boolean isProposedValue() {
    return proposedValue;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    ProposeValue that = (ProposeValue) o;

    return proposedTry == that.proposedTry && proposedValue == that.proposedValue;
  }

  @Override
  public int hashCode() {
    return Objects.hash(proposedTry, proposedValue);
  }

  @Override
  public String toString() {
    return String.format(
        "ProposeValue{proposed_try=%s, proposed_value=%b}", proposedTry, proposedValue);
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/consensus/ProposeValue.kt
package com.github.dedis.popstellar.model.network.method.message.data.consensus

import com.github.dedis.popstellar.model.Immutable
import com.google.gson.annotations.SerializedName
import java.util.Objects

@Immutable
class ProposeValue(
    @field:SerializedName("proposed_try") val proposedTry: Int,
    @field:SerializedName("proposed_value") val isProposedValue: Boolean
) {

  override fun equals(other: Any?): Boolean {
    if (this === other) {
      return true
    }
    if (other == null || javaClass != other.javaClass) {
      return false
    }
    val that = other as ProposeValue
    return proposedTry == that.proposedTry && isProposedValue == that.isProposedValue
  }

  override fun hashCode(): Int {
    return Objects.hash(proposedTry, isProposedValue)
  }

  override fun toString(): String {
    return "ProposeValue{proposed_try=$proposedTry, proposed_value=$isProposedValue}"
  }
}
